iOS_SDK/OneSignalSDK/OneSignalCore/Source/API/OneSignalRequest.m

```diff
@@ -89,7 +89,14 @@ -(NSMutableURLRequest *)urlRequest {
 }
 
 -(BOOL)missingAppId {
-    return self.parameters[@"app_id"] == nil || [self.parameters[@"app_id"] length] == 0;
+
+  if ([self.path containsString:@"apps/"]) {
+    NSArray *pathComponents = [self.path componentsSeparatedByString:@"/"];
+    NSUInteger x = [pathComponents indexOfObject:@"apps"] + 1; // Find the index that follows "apps" in the path
+    return ([pathComponents count] < x || [[pathComponents objectAtIndex:x] length] == 0);
```

Comment:
What situation does `[pathComponents count] < x` account for? I think this would always be `false`.

Thinking about an example `@"apps/%@/"` ...
When the request is created, most of the time the `app_id` is injected by string interpolation, and you account for when it's the empty string `@"apps//"`. I think it can be likely that the app_id is actually `nil` in which case the path is `@"apps/(null)/"`

[[pathComponents objectAtIndex:x]  isEqual: @"(null)"]

Reply:
I agree with changing the method itself than overriding `missingAppId` in the live activity requests, since the new user model requests will also have app_id in the path, not params.

Reply:
Good catch nan, thanks `[[pathComponents objectAtIndex:x] isEqual: @"(null)"]`